lib/rspec/mocks/targets.rb

```diff
@@ -101,6 +101,14 @@ def expression
 
     # @private
     class AnyInstanceAllowanceTarget < TargetBase
+      def initialize(klass)
+        super
+        return unless RSpec::Mocks.configuration.verify_partial_doubles?
+        RSpec::Mocks.configuration.verifying_double_callbacks.each do |block|
+          block.call(ObjectReference.for(klass))
+        end
```

Comment:
This will cause the callback to be executed for every invocation of `allow_any_instance_of` which is different to their existing behaviour (their use is memoized to be once per class thanks to the space collection), I think tis should be place in `AnyInstance::Recorder#initialize` instead which is the same pattern as the normal proxies...